Sources/WeScan/Scan/CaptureSessionManager.swift

```diff
@@ -141,19 +141,30 @@ final class CaptureSessionManager: NSObject, AVCaptureVideoDataOutputSampleBuffe
 
         switch authorizationStatus {
         case .authorized:
-            DispatchQueue.main.async {
-                self.captureSession.startRunning()
+            DispatchQueue.global(qos: .userInitiated).async { [weak self] in
+                self?.captureSession.startRunning()
+                DispatchQueue.main.async {
+                    self?.isDetecting = true
+                }
             }
-            isDetecting = true
         case .notDetermined:
-            AVCaptureDevice.requestAccess(for: AVMediaType.video, completionHandler: { _ in
-                DispatchQueue.main.async { [weak self] in
-                    self?.start()
+            AVCaptureDevice.requestAccess(for: AVMediaType.video, completionHandler: { [weak self] granted in
+                if granted {
+                    DispatchQueue.main.async {
+                        self?.start()
+                    }
+                } else {
+                    DispatchQueue.main.async {
```

Comment:
Nit: I suppose this could to with one `main.async` and the if-else can live within it

Reply:
Thank you for your suggestion, BasThomas. You're right that consolidating the DispatchQueue.main.async calls and nesting the if-else statement within it would indeed reduce redundancy and improve the readability of the code.